tensorflow/compiler/mlir/lite/transforms/optimize.cc

```diff
@@ -206,6 +206,15 @@ DenseElementsAttr GetShape(Value output_val) {
       llvm::makeArrayRef(shape));
 }
 
+bool notFromQuantOpDifferentQuanteUse( Value val, TypeAttr qtype) {
+  auto val_defn_op = val.getDefiningOp();
```

Comment:
remove the extra space.

Reply:
Tensorflow's C++ style requires `NotFromQuantOpDifferentQuanteUse`

Reply:
I think this check is too restrict...we just need to check the quantization parameters are the same:
```
return val.getType().cast<ShapedType>().getElementalType() == qtype.getValue();
```

Reply:
Hmm... to ignore shape when checking for identical quantization we presumably need to compare with ElementType of qtype attr value.  

Explanation: quant::GetQuantizedTypeAttr appears to constructs TFL::QuantizeOp qtype attr as a ShapedType if (as would typically be the case) the output of the original fake_quant op(s) was shaped (vector, tensor ...).  

Also can we exclude   fake_quant input being a non ShapedType type?   The construction of qtype attr type seems to go to a lot of trouble to handle non-ShapedType cases.... 
Safer to handle the non-ShapedType cases.

I'll push a corresponding change for review.